### PR TITLE
DOCS-7387 Container Cost Allocation Doc Update

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Add tabs for Azure, AWS, and Google to nest sections of information on the Container Cost Allocations page.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/cloud_cost_management/_index.md [hunk 1]
```diff
@@ -43,16 +43,23 @@ Datadog ingests your cloud cost data and transforms it into queryable metrics. I
...
-- Export timeseries costs data using the [Metrics API][1].
+  
+{{< img src="cloud_cost/cloud_cost_data_source.png" alt="Cloud Cost available as a data source in dashboard widget creation" style="width:60%;" >}}
+
+- Export a timeseries graph of costs data using the [Metrics API][1].
+
+## Allocate costs
 
-{{< img src="cloud_cost/cloud_cost_data_source.png" alt="Cloud Cost available as a data source in dashboard widget creation" >}}
+Use [Container Cost Allocation][2] metrics to identify Kubernetes, AWS ECS, Azure, and Google Cloud cluster and workload costs, and gain visibility into into pod-level costs, idle resource costs, costs by resource type, and more.
```
Comment (alai97) [inline content/en/cloud_cost_management/_index.md:58]:
```suggestion
Use [Container Cost Allocation][2] metrics to discover costs associated with clusters and workloads across Kubernetes, AWS ECS, Azure, and Google Cloud. Gain visibility into pod-level costs, identify idle resource costs, and analyze costs by resource type.
```

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 2]
```diff
@@ -16,56 +17,125 @@ further_reading:
...
+
+CCM allocates costs of AWS EC2 instances to your AWS ECS clusters.
+
+The following table presents the list of collected features and the minimal Agent and Cluster Agent versions for each.
+
+| Feature | Minimal Agent version | Minimal Cluster Agent version |
+|---|---|---|
+| Container Cost Allocation | 7.27.0 | 1.11.0 |
+| AWS Persistent Volume Allocation | 7.46.0 |  |
+| AWS Split Cost Allocation | TBD | |
```
Comment (alai97) [inline content/en/cloud_cost_management/container_cost_allocation.md:65]: @elihschiff can you add the minimal Agent version for this feature? 
```suggestion
| AWS Split Cost Allocation | TBD | |
```

**Review by elihschiff — COMMENTED** [review]
I was only able to get through so much of the doc today, I will take another look tomorrow and leave more comments but this is what I have so far.

I think the structure is a lot better! I mostly left comments about the content which you obviously have less direct experience with than I do but overall it was pretty good!


Overall question: Is there any way to make AWS the default choice when you load up the page? It is the cloud we support the most features on and the cloud we have the most customers in. Maybe put the cloud tabs in alphabetical order?

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 3]
```diff
@@ -75,59 +145,169 @@ Based on resources the task has used, Datadog assigns the appropriate portion of
...
 
-- Usage: Cost of memory and CPU used by workloads, based on the average usage on that day.
-- Workload idle: Cost of memory and CPU that is being reserved and allocated but not used. This is the difference between the total resources requested and the average usage.
-- Cluster idle: Cost of memory and CPU not reserved by workloads in a cluster. This is the difference between the total cost of the resources and what is allocated to workloads.
+Costs are allocated into the following spend types:
 
-### Persistent volumes (AWS only)
+| Spend type | Description    |
+| -----------| -----------    |
+| Managed service fee | Cost of associated fees charged by the provider for managing the cluster, such as AWS ECS fees. |
```
Comment (elihschiff) [inline content/en/cloud_cost_management/container_cost_allocation.md:220]: Managed service fee should probably be in their own section unrelated to compute costs (this goes for all 3 clouds)
I think we should move managed service fees up to a type of resource personally. I would put it as the last resource in the list for each cloud.

Also small typo for each cloud the such as should be: AWS -> EKS, Azure -> AKS, GCP -> GKE

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 4]
```diff
@@ -75,59 +145,169 @@ Based on resources the task has used, Datadog assigns the appropriate portion of
...
+| CPU | {{< X >}} | {{< X >}} | {{< X >}} |
+| Memory | {{< X >}} | {{< X >}} | {{< X >}} |
+| Persistent Volumes | {{< X >}} |  |  |
+| Managed service fees | {{< X >}} | {{< X >}} | {{< X >}} |
+| ECS costs | {{< X >}} | N/A | N/A |
+| Networking costs |  | Limited* | Limited* |
+| GPU |  |  | Limited* |
+| Local storage |  | Limited* | Limited* |
+
+For `Limited*` resources, you are required to tag them as part of your Kubernetes cost allocation using the `allocated_spend_type:<resource>_not_supported` tagging format, for example: `allocated_spend_type:gpu_not_supported`. These resources are host-level costs, not pod or namespace-level costs.
```
Comment (elihschiff) [inline content/en/cloud_cost_management/container_cost_allocation.md:273]:
```suggestion
`Limited*` resources have been identified to be a part of your Kubernetes spend, but are not currently fully allocated to specific workloads or pods. These resources are host-level costs, not pod or namespace-level costs. You will see `allocated_spend_type:<resource>_not_supported` on these resources. 
```

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 5]
```diff
@@ -75,59 +145,169 @@ Based on resources the task has used, Datadog assigns the appropriate portion of
...
+The cost of a host instance is split into two components: 60% for the CPU and 40% for the memory. Each component is allocated to individual workloads based on their resource reservations and usage.
+
+Costs are allocated into the following spend types:
+
+| Spend type | Description    |
+| -----------| -----------    |
+| Managed service fee | Cost of associated fees charged by the provider for managing the cluster, such as GKE fees. |
+| Usage | Cost of resources (such as memory and CPU) used by workloads, based on the average usage on that day. |
+| Workload idle | Cost of resources (such as memory and CPU) that are reserved and allocated but not used by workloads. This is the difference between the total resources requested and the average usage. |
+| Cluster idle | Cost of resources (such as memory and CPU) that are not reserved by workloads in a cluster. This is the difference between the total cost of the resources and what is allocated to workloads. |
```
Comment (elihschiff) [inline content/en/cloud_cost_management/container_cost_allocation.md:253]:
```suggestion
| Cluster idle | Cost of resources (such as memory and CPU) that are not reserved by workloads in a cluster. This is the difference between the total cost of the resources and what is allocated to workloads. |
| Not monitored | Cost of resources where the spend type is unknown. Installing the datadog agent on these clusters/nodes will resolve the issue |
```
For GCP only: we need a `Not monitored` section that pretty much tells users that they need to install the datadog agent to get the full support here. I am not sure about my wording though. Feel free to update it

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 6]
```diff
@@ -146,7 +326,7 @@ Datadog consolidates and applies additional tags from various sources to cost me
```
Comment (elihschiff) [inline content/en/cloud_cost_management/container_cost_allocation.md:326]: In the above table for `Out-of-the-box tag` I think we should rework it to just be a table that describes the `orchestrator` tag. (dropping all the other rows as they are becoming redundant with the rest of the page)

The orchestrator section should say something about how we add a `orchestrator` tag to all resources associated with your cluster. 

If there could be a tab per cloud here as well. Azure and GCP will have the table with a single row for `orchestrator:kubernetes` while AWS will also have the `orchestrator:ecs` row

**Review by elihschiff — COMMENTED** [review]
One thing I noticed is that linking to specific parts of the page under tabs does not work so well, not sure if we can fix this or not?

Going to this link in a new tab does not bring you to the correct section for example https://docs-staging.datadoghq.com/alai97/update-ccm-container-cost-allocation-docs/cloud_cost_management/container_cost_allocation/?tab=aws#compute

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 7]
```diff
@@ -16,56 +17,125 @@ further_reading:
...
-     - Ensure that you enable the [**Orchestrator Explorer**][6] in your Agent configuration.
-     - Container cost allocation requires **Agent version >= 7.27.0** and **Cluster Agent version >= 1.11.0**.
-     - AWS Persistent Volume allocation requires **Agent version >= 7.46.0**
-3. For AWS ECS support, set up [**Datadog Container Monitoring**][3] in ECS tasks.
-     - Optionally, enable [AWS Split Cost Allocation][5] for usage-based ECS allocation.
+{{< tabs >}}
+{{% tab "Azure" %}}
 
-## Cost allocation
+CCM allocates costs of Azure VM instances to your Azure Kubernetes Service (AKS) clusters.
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/cloud_cost_management/container_cost_allocation.md:39]:
```suggestion
CCM allocates costs of all Kubernetes clusters, including those managed through Azure Kubernetes Service (AKS). 
```

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 8]
```diff
@@ -16,56 +17,125 @@ further_reading:
...
+| Container Cost Allocation | 7.27.0 | 1.11.0 |
+
+[101]: https://app.datadoghq.com/cost/setup
+[102]: /containers/kubernetes/installation/?tab=operator
+[103]: /infrastructure/containers/orchestrator_explorer?tab=datadogoperator
+
+{{% /tab %}}
+{{% tab "AWS" %}}
+
+CCM allocates costs of AWS EC2 instances to your AWS ECS clusters.
```
[the remark this answers is not in the document]
Comment (alai97) [reply content/en/cloud_cost_management/container_cost_allocation.md:57]:
```suggestion
CCM allocates costs of all AWS EC2 instances across your container orchestration environments, including Kubernetes clusters and AWS-specific ECS clusters.
```

**Review by elihschiff — APPROVED** [review]
I shared this around my team so lets wait a bit and see if anyone else has any feedback otherwise we can just merge it

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 9]
```diff
@@ -16,56 +17,108 @@ further_reading:
...
-Next, Datadog looks at all of the pods running on that node for the day. The cost of the node is allocated to the pod based on the resources it has used and the length of time it ran. This calculated cost is enriched with all of the pod's tags.
+{{% /tab %}}
+{{% tab "Azure" %}}
 
-**Note**: Only _tags_ from pods and nodes are added to cost metrics. To include labels, enable labels as tags for [nodes][7] and [pods][8].
+CCM allocates costs of all Kubernetes clusters, including those managed through Azure Kubernetes Service (AKS). 
 
-#### Persistent volume storage (AWS only)
+1. Configure the Azure Cost Management integration on the [Cloud Costs Setup page][101].
+1. Install the [**Datadog Agent**][102] in a Kubernetes environment and ensure that you enable the [**Orchestrator Explorer**][103] in your Agent configuration.
```
Comment (elihschiff) [inline content/en/cloud_cost_management/container_cost_allocation.md:63]: Very small nit, for AWS and GCP this comes after the version table below but in Azure it comes before

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 10]
```diff
@@ -16,56 +17,108 @@ further_reading:
...
+| Container Cost Allocation | 7.27.0 | 1.11.0 |
+
+[101]: https://app.datadoghq.com/cost/setup
+[102]: /containers/kubernetes/installation/?tab=operator
+[103]: /infrastructure/containers/orchestrator_explorer?tab=datadogoperator
+
+{{% /tab %}}
+{{% tab "Google" %}}
+
+CCM allocates costs of Google Kubernetes Engine instances to your Google Cloud clusters.
```
Comment (elihschiff) [inline content/en/cloud_cost_management/container_cost_allocation.md:78]:
```suggestion
CCM allocates costs of all Kubernetes clusters, including those managed through Google Kubernetes Engine (GKE).
```

Making this phraising the same as what we have for Azure as I think it is more clear

File: content/en/cloud_cost_management/container_cost_allocation.md [hunk 11]
```diff
@@ -16,56 +17,108 @@ further_reading:
...
-     - Ensure that you enable the [**Orchestrator Explorer**][6] in your Agent configuration.
-     - Container cost allocation requires **Agent version >= 7.27.0** and **Cluster Agent version >= 1.11.0**.
-     - AWS Persistent Volume allocation requires **Agent version >= 7.46.0**
-3. For AWS ECS support, set up [**Datadog Container Monitoring**][3] in ECS tasks.
-     - Optionally, enable [AWS Split Cost Allocation][5] for usage-based ECS allocation.
+{{< tabs >}}
+{{% tab "AWS" %}}
 
-## Cost allocation
+CCM allocates costs of all AWS EC2 instances across your container orchestration environments, including Kubernetes clusters and AWS-specific ECS clusters.
```
Comment (elihschiff) [inline content/en/cloud_cost_management/container_cost_allocation.md:37]:
```suggestion
CCM allocates costs of AWS ECS clusters as well as all Kubernetes clusters, including those managed through Elastic Kubernetes Service (EKS).
```

Making the phrasing here more consistent across the 3 clouds

**Review by elihschiff — COMMENTED** [review]
Adding a few other very small NITs I noticed. Nothing major

Another small NIT: Something is weird about the sidebar colors not sure why so may sections turn purple depending on where you are on the page it does not seem consistent 

image

**Review by enemyghost — APPROVED** [review]

**Pull request merged**